src/reusableState/customList/tests/StringListReducers.spec.ts

```diff
@@ -0,0 +1,142 @@
+import {IReduxAction} from '../../../utils/ReduxUtils';
+import {
+    addStringList,
+    addValueStringList,
+    IStringListPayload,
+    removeStringList,
+    removeValueStringList,
+    StringListActions,
+} from '../StringListActions';
+import {IStringListCompositeState, stringListCompositeState, stringListInitialState} from '../StringListReducers';
+
+describe('StringList', () => {
+
+    describe('StringList Reducers', () => {
+
+        const stateId: string = 'test';
+        const oldState: IStringListCompositeState = {
+            [stateId]: {
+                id: stateId,
+                list: [],
+            },
+        };
+
+        const genericAction: IReduxAction<IStringListPayload> = {
+            type: 'pokeball',
+        };
+
+        it('should return the default state if the action has no payload and the state is undefined', () => {
+            const newState: IStringListCompositeState = stringListCompositeState(undefined, genericAction);
+
+            expect(newState).toEqual(stringListInitialState);
+            expect(newState).not.toBe(stringListInitialState);
+        });
+
+        it(`should return the default state if the payload id is not in the state, the action type is not ${StringListActions.add} and the state is undefined`, () => {

```

Comment:
no variables in test description, it will make the test unsearchable

Reply:
same for ut below

Reply:
Im not complete agree with that, if we rework the name of the variable, the test name will be wrong after that. We can search the test name with the uts structure easily.